### ISSUE TITLE
react-native run-{ios,android} fails with Process terminated on launchPackager in 0.58.0-rc.3

### ISSUE THREAD

**Issue by msand** [facebook/react-native#23136]
## Environment
  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.6
      CPU: (4) x64 Intel(R) Core(TM) i5-4278U CPU @ 2.60GHz
      Memory: 327.33 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.2.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
      Android SDK:
        API Levels: 16, 23, 24, 25, 26, 27, 28
        Build Tools: 23.0.1, 25.0.0, 25.0.1, 25.0.2, 25.0.3, 26.0.1, 26.0.2, 26.0.3, 27.0.1, 27.0.2, 27.0.3, 28.0.2, 28.0.3
        System Images: android-16 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.1 AI-173.4819257
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.3 => 16.6.3 
      react-native: 0.58.0-rc.3 => 0.58.0-rc.3 
    npmGlobalPackages:
      create-react-native-app: 2.0.2
      react-native-cli: 2.0.1
      react-native-create-library: 3.1.2
      react-native-git-upgrade: 0.2.7
      react-native-init: 0.6.7

## Description
`react-native run-{ios,android}` and starting from xcode fails in a clean react-native init project with latest 0.58.0-rc.3
metro fails with `Process terminated`
running `react-native start` allows metro to start

## Reproducible Demo
react-native init --version="0.58.0-rc.3" MyNewApp
cd MyNewApp
react-native run-ios

**Comment by pjay79** [comment]
Same issue here. Running react-native start followed by react-native run-ios / android works.

**Comment by aditiadika** [comment]
same issue here. and run-react native start its not working too

**Comment by vshab** [comment]
The 0.58 release is affected as well.

**Comment by LinMaris** [comment]
me too, 0.58 release doesn't seem to work.
I just
``` 
react-native init DemoProject
cd DemoProject/
react-native run-ios
``` 
That always appeared this
![image](https://user-images.githubusercontent.com/17720805/51738159-6bc37e00-20c9-11e9-89cd-fc2db2aa2e88.png)

**Comment by eshikerya** [comment]
works if metro started from ```nuclide```, but fails from ```run-{iOS,android}```

**Comment by msand** [comment]
seems fixed in 0.58.1 https://github.com/facebook/react-native/pull/23160

**Comment by kelset** [comment]
Yeah we fixed this in 0.58.1 - closing this issue for now, if it resurfaces I'll reopen ;)
